src/index.js

```diff
@@ -27,7 +28,13 @@ async function timeRegions({ file, sources, regions }, iter) {
             tileSize,
             extent,
           });
-          const p = tileCoords.map(t => timeGetTile(loader, { ...t, loaderSelection }));
+          const queue = new PQueue({ concurrency: 10 });
+          const p = []
+          for (const t of tileCoords) {
+            p.push(queue.add(() => timeGetTile(loader, { ...t, loaderSelection })));
+          }
+          // guarantees that all work on queue has finished
+          await queue.onIdle();
           const times = await Promise.all(p); // await all requests
```

Comment:
```suggestion
          const p = tileCoords.map(t => timeGetTile(loader, { ...t, loaderSelection }));
          const queue = new PQueue({ concurrency: 10 });
          const records = []
          for (const t of tileCoords) {
            queue.add(async () => {
            	const [start, end] = await timeGetTile(loader, { ...t, loaderSelection });
				const { x, y, z } = t;
				const record = {
                  iter,
                  file,
                  format,
                  compression,
                  // The zoom level is the z, not the provided one.
                  zoom: -z,
                  tileSize,
                  regionId: id,
                  yCoord,
                  xCoord,
                  height,
                  width,
                  numChannels: n,
                  tileId: `${x}-${y}-${z}`,
                  startTime: start,
                  endTime: end,
                };
                records.push(record);
            );
          }
          // guarantees that all work on queue has finished
          await queue.onIdle();

Reply:
I don't think queue.add() returns anything, so you need to capture the records as they are executed.

Reply:
and then just await the queue when it's finished

Reply:
Sorry, it seemed from the documentation that it returned a promise, but this is cleaner.

Reply:
Ah, it does return a Promise, but I'm not sure if it resolves to a value. So we need to collect the data somehow.

Reply:
It seemed like it was just resolving to the return value, but this is cleaner nonetheless.  I updated it.